src/Testcontainers/Builders/TestcontainersCloudEndpointAuthenticationProvider.cs

```diff
@@ -0,0 +1,127 @@
+  using DotNet.Testcontainers.Images;
+  using JetBrains.Annotations;
+
+  /// <inheritdoc cref="IDockerRegistryAuthenticationProvider" />
+  [PublicAPI]
+  internal sealed class TestcontainersHostEndpointAuthenticationProvider : DockerEndpointAuthenticationProvider, ICustomConfiguration
+  {
+    private readonly ICustomConfiguration _customConfiguration = new TestcontainersHostConfiguration();
+
+    private readonly Uri _dockerEngine;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="TestcontainersHostEndpointAuthenticationProvider" /> class.
+    /// </summary>
+    public TestcontainersHostEndpointAuthenticationProvider()
+    {
+      _dockerEngine = GetDockerHost();
+    }
+
+    /// <inheritdoc />
+    public override bool IsApplicable()
+    {
+      return _dockerEngine != null && "tcp".Equals(_dockerEngine.Scheme, StringComparison.OrdinalIgnoreCase);
+    }
+
+    /// <inheritdoc />
+    public override IDockerEndpointAuthenticationConfiguration GetAuthConfig()
+    {
+      return new DockerEndpointAuthenticationConfiguration(_dockerEngine);
+    }
+
+    /// <inheritdoc />
+    public string GetDockerConfig()
+    {
+      return _customConfiguration.GetDockerConfig();
+    }
+
+    /// <inheritdoc />
+    public Uri GetDockerHost()
+    {
+      return _customConfiguration.GetDockerHost();
+    }
+
+    /// <inheritdoc />
+    public string GetDockerHostOverride()
+    {
+      return _customConfiguration.GetDockerHostOverride();
+    }
+
+    /// <inheritdoc />
+    public string GetDockerSocketOverride()
+    {
+      return _customConfiguration.GetDockerSocketOverride();
+    }
+
+    /// <inheritdoc />
+    public JsonDocument GetDockerAuthConfig()
+    {
+      return _customConfiguration.GetDockerAuthConfig();
+    }
+
+    /// <inheritdoc />
+    public string GetDockerCertPath()
+    {
+      return _customConfiguration.GetDockerCertPath();
+    }
+
+    /// <inheritdoc />
+    public bool GetDockerTls()
+    {
+      return _customConfiguration.GetDockerTls();
+    }
+
+    /// <inheritdoc />
+    public bool GetDockerTlsVerify()
+    {
+      return _customConfiguration.GetDockerTlsVerify();
+    }
+
+    /// <inheritdoc />
+    public bool GetRyukDisabled()
+    {
+      return _customConfiguration.GetRyukDisabled();
+    }
+
+    /// <inheritdoc />
+    public bool GetRyukContainerPrivileged()
+    {
+      return _customConfiguration.GetRyukContainerPrivileged();
+    }
+
+    /// <inheritdoc />
+    public IImage GetRyukContainerImage()
+    {
+      return _customConfiguration.GetRyukContainerImage();
+    }
+
+    /// <inheritdoc />
+    public string GetHubImageNamePrefix()
+    {
+      return _customConfiguration.GetHubImageNamePrefix();
+    }
+
+    private sealed class TestcontainersHostConfiguration : PropertiesFileConfiguration
+    {
+      protected override Uri GetDockerHost(string propertyName)
+      {
+        return base.GetDockerHost("testcontainers.host");
```

Comment:
```suggestion
        return base.GetDockerHost("tcc.host");
```

Haven't yet discussed, but maybe better to explicitly use a `tcc` namespace, to keep the `testcontainers` namespace potentially free.

Reply:
Sorry for the confusion and to revert this. Going forward, this abstraction seems to be useful (and there can be a Testcontainers host that is `!=` running in TCC), so using `tc.host` might be the better way (see tc-node: https://github.com/testcontainers/testcontainers-node/pull/564/files#diff-a216e10441367a5e1ff574ed8c1b678a8a80886b31c6029231cdc291cb3e8f10R34)